phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -184,7 +184,7 @@ public static ColumnResolver getResolverForCompiledDerivedTable(PhoenixConnectio
             Expression sourceExpression = projector.getColumnProjector(column.getPosition()).getExpression();
             PColumnImpl projectedColumn = new PColumnImpl(column.getName(), column.getFamilyName(),
                     sourceExpression.getDataType(), sourceExpression.getMaxLength(), sourceExpression.getScale(), sourceExpression.isNullable(),
-                    column.getPosition(), sourceExpression.getSortOrder(), column.getArraySize(), column.getViewConstant(), column.isViewReferenced());
+                    column.getPosition(), sourceExpression.getSortOrder(), column.getArraySize(), column.getViewConstant(), column.isViewReferenced(), null);
```

Comment:
This is like a copy constructor, so instead of null for the new arg, you should likely use column.getExpressionStr().